packages/components/src/components/text/text.jsx

```diff
@@ -0,0 +1,46 @@
+import classNames from 'classnames';
+import React from 'react';
+
+const Text = ({ children, size, color, align, weight, lineHeight, className }) => (
+    <>
+        <p
+            className={classNames('text', className, {
```

Comment:
```jsx
const Text = ({ children, size, color, align, weight, lineHeight, className }) =>  {

const getTextSize = () => {
      switch (size): 
          case 'xsmall': 
              return 'var(--text-size-xs)';
      // ...
}
return (
        <p
            className={classNames('text', className)
            style={{
                '--text-size': getTextSize()
            // ...
```

We can apply the same technique to the rest of component's properties, rinse and repeat. 

Suggestion: With a better naming, we can even use string manipulation instead of switch/case to detect variable name.